### DISCUSSION TITLE
BOM parts stock location

### DISCUSSION THREAD

**Discussion by tnemrap** [inventree/InvenTree#6082] [Q&A]
I am working on a picklist of the BOM.

![image](https://github.com/inventree/InvenTree/assets/1627294/2256cbc7-ca38-4ebd-846d-2124115ab4d0)

I am wondering how to get stock location from line.sub_part object?

**Comment by tnemrap** [comment]
I find this
{{ line.sub_part.stock_items.all.0.location.name }}
using 0 just to get the first stock_item

![image](https://github.com/inventree/InvenTree/assets/1627294/02cf44bf-df74-4c35-9b02-c5f851bce533)

**Comment by LavissaWoW** [comment]
The logic separating a part and its BOM, and Stock Items, makes it such that there's no deterministic way to add a Stock Location to a BOM report that's based on Stock Items.

Say you have one in-house stock, and one external stock. And you have items in both of these.
Trying to fetch this in a BOM report would yield one of those, but that wouldn't guarantee it's the one you want.

Rather than making a PickList for a BOM report, I'd suggest making a PickList for a Build Report. 
In a Build Report, you have access to the actual Stock Items that have been allocated to the Build Order, along with their Stock Locations.

**Reply by tnemrap** [reply]
I will look to move it to build order.
